src/settings.rs

```diff
@@ -59,36 +41,40 @@ impl Settings {
 
         s.try_deserialize()
     }
+
+    /// Creates a new `Settings` instance from the given command line arguments.
+    pub fn from_args(args: CmdLineArgs) -> Result<Self, ConfigError> {
+        let (manager_name, manager_rpc_srv_addr) = args
+            .manager_server
+            .split_once('@')
+            .expect("cannot get information of the manager server");
```

Comment:
I'd appreciate if you could kindly refer to our guideline & convention on panicking and expect.

Reply:
Fixed not to panic but return Err.
Also fixed `expect` in `default_config_builder` function to state 'verified by
const'